curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -277,24 +307,33 @@ void DiskCacheManager::TrimCache() {
                     if (iter == cachedObjName_.end()) {
                         VLOG(6) << "remove disk file error"
                                 << ", file is not exist in cachedObjName"
-                                << ", file = " << *cacheKeyIter;
+                                << ", file is: " << *cacheKeyIter;
                         cachedObjNameTmp.erase(cacheKeyIter);
                         continue;
                     }
                     cachedObjName_.erase(iter);
                 }
+                struct stat statReadFile;
+                ret = posixWrapper_->stat(cacheReadFile.c_str(), &statReadFile);
+                if (ret != 0) {
+                    VLOG(3) << "remove disk file error"
+                               << ", file is: " << *cacheKeyIter;
+                    continue;
+                }
                 // if remove disk file before delete cache,
                 // then read maybe fail.
                 const char *toDelFile;
                 toDelFile = cacheReadFile.c_str();
                 ret = posixWrapper_->remove(toDelFile);
                 if (ret < 0) {
                     LOG(ERROR)
-                        << "remove disk file error, file = " << *cacheKeyIter;
+                        << "remove disk file error, file is: " << *cacheKeyIter;
                     cachedObjNameTmp.erase(cacheKeyIter);
                     continue;
                 }
-                VLOG(3) << "remove disk file success, file = " << *cacheKeyIter;
+                DecDiskUsedBytes(statReadFile.st_size);
```

Comment:
Is it possible that statReadFile has holes?   SetDiskInitUsedBytes use “du”  to calc disk used， which is not conclude the size of hole？

Reply:
the file has no hole